### PR TITLE
refactor: use sqlalchemy engine to write and read dataframes by default

### PR DESCRIPTION
**Pull request by chamini2**
adapters to test:
- bigquery
- snowflake
- postgres

### CODE REVIEW

File: src/faldbt/lib.py [hunk 1]
```diff
@@ -499,6 +502,18 @@ def null_dump(sql, *multiparams, **params):
     return sqlalchemy.create_mock_engine(url_string, executor=null_dump)
 
 
+def _create_engine_from_connection(adapter: SQLAdapter):
+    if adapter.type() == "postgres":
+        url_string = "postgresql+psycopg2://"
+    else:
+        # TODO: add special cases as needed
+        logger.warn("No explicit url string for adapter {}", adapter.type())
+        url_string = f"{adapter.type()}://"
```
Comment (isidentical) [inline src/faldbt/lib.py:511]: Can we use `adapter.credentials.connection_string` here (since it would also contain the prefix)?

**Review by isidentical — APPROVED** [review]
Looks great, thanks! Quick question below

**Review by mederka — APPROVED** [review]

File: src/faldbt/lib.py (hunk 1 above)
Comment (chamini2) [reply src/faldbt/lib.py:511]: `adapter.credentials.connection_string` is this a thing?!

Comment (isidentical) [reply src/faldbt/lib.py:511]: I recall seeing it somewhere, but now looking deeply I can't find it either. Probably misremembered :/

**Pull request merged**